frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx

```diff
@@ -0,0 +1,711 @@
+import { screen, waitFor, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { createMockDashboard } from "metabase-types/api/mocks";
+import { getBrokenUpTextMatcher } from "__support__/ui";
+
+import { getMockResource, setup } from "./setup";
+
+describe("Static Embed Setup phase", () => {
+  describe.each([
+    {
+      resourceType: "dashboard" as const,
+    },
+    {
+      resourceType: "question" as const,
+    },
+  ])("$resourceType", ({ resourceType }) => {
+    describe("EmbedModalContentStatusBar", () => {
+      it(`should render actions banner for non-published ${resourceType}`, () => {
+        const { onUpdateEnableEmbedding } = setup({
+          props: {
+            resourceType,
+          },
+        });
+
+        expect(
+          screen.getByText(
+            `You will need to publish this ${resourceType} before you can embed it in another application.`,
+          ),
+        ).toBeVisible();
+
+        const button = screen.getByRole("button", {
+          name: "Publish",
+        });
+        expect(button).toBeVisible();
+
+        userEvent.click(button);
+
+        expect(onUpdateEnableEmbedding).toHaveBeenLastCalledWith(true);
+      });
+
+      it(`should render actions banner for published ${resourceType}`, () => {
+        const { onUpdateEnableEmbedding } = setup({
+          props: {
+            resource: getMockResource(resourceType, true),
+            resourceType,
+          },
+        });
+
+        expect(
+          screen.getByText(
+            `This ${resourceType} is published and ready to be embedded.`,
+          ),
+        ).toBeVisible();
+
+        const button = screen.getByRole("button", {
+          name: "Unpublish",
+        });
+        expect(button).toBeVisible();
+
+        userEvent.click(button);
+
+        expect(onUpdateEnableEmbedding).toHaveBeenLastCalledWith(false);
+      });
+    });
+
+    describe("Overview tab", () => {
+      it("should render content", () => {
+        setup({
+          props: {
+            resourceType,
+          },
+          activeTab: "Overview",
+        });
+
+        expect(screen.getByText("Setting up a static embed")).toBeVisible();
+
+        expect(
+          screen.getByText(
+            `To embed this ${resourceType} in your application you’ll just need to publish it, and paste these code snippets in the proper places in your app.`,
+          ),
+        ).toBeVisible();
+
+        const link = screen.getByRole("link", {
+          name: "documentation",
+        });
+        expect(link).toBeVisible();
+        expect(link).toHaveAttribute(
+          "href",
+          "https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=oss&utm_media=static-embed-settings-overview",
+        );
+
+        if (resourceType === "dashboard") {
+          // eslint-disable-next-line jest/no-conditional-expect
```

Comment:
I'm not really a fan of going against this rule, but the file is probably already to big as it is so let's see what others have to say

Reply:
I agree. I think if we have to have conditionals within an `each` clause in a test, then we probably should split it out into somewhere else